src/aten/sycl/EmbeddingBag.cpp

```diff
@@ -0,0 +1,682 @@
+        accscalar_t,                                                        \
+        index_t>(                                                           \
+        mode,                                                               \
+        input,                                                              \
+        offset,                                                             \
+        weight,                                                             \
+        output,                                                             \
+        offset2bag,                                                         \
+        bag_size,                                                           \
+        per_sample_weights_defined,                                         \
+        per_sample_weights,                                                 \
+        per_sample_weights_stride,                                          \
+        max_indices,                                                        \
+        WGNumber,                                                           \
+        numBags,                                                            \
+        weight_total_elem,                                                  \
+        chunk_size,                                                         \
+        bag_chunk_num,                                                      \
+        bag_wi_num,                                                         \
+        bagsPerLoop,                                                        \
+        input_length,                                                       \
+        weight_stride0,                                                     \
+        weight_stride1,                                                     \
+        include_last_offset,                                                \
+        padding_idx,                                                        \
+        if_align_vector);                                                   \
+    sycl_kernel_submit(                                                     \
+        global_range, local_range, getCurrentSYCLQueue(), caller);          \
+  };
+
+  switch (vec_size) {
+    case 16: {
+      VEC_EMBBAG_KERNEL(16);
+      break;
+    }
+    case 8: {
+      VEC_EMBBAG_KERNEL(8);
+      break;
+    }
+    case 4: {
+      VEC_EMBBAG_KERNEL(4);
+      break;
+    }
+    case 2: {
+      VEC_EMBBAG_KERNEL(2);
+      break;
+    }
+    case 1: {
+      VEC_EMBBAG_KERNEL(1);
+      break;
+    }
+    default:
+      TORCH_INTERNAL_ASSERT(
+          false,
+          "Unexpected vectorization size for EmbeddingBag. vec size ",
+          vec_size);
+  }
+#undef VEC_EMBBAG_KERNEL
+}
+
+} // namespace
+
+std::tuple<Tensor, Tensor, Tensor, Tensor> _embedding_bag_kernel(
+    const Tensor& weight_t,
+    const Tensor& indices_t,
+    const Tensor& offsets_t,
+    const bool scale_grad_by_freq,
```

Comment:
Don't we support the semantics in kernel? Please review CUDA kernel implementation to get it supported.

Reply:
Checked. Not used in forward.